mozilla-release/mobile/android/base/java/org/mozilla/gecko/activitystream/homepanel/ActivityStreamPanel.java

```diff
@@ -290,22 +292,52 @@ public void onTabChanged(Tab tab, Tabs.TabEvents msg, String data) {
         // existing users are shown 'customize_newtab_snackbar' which the user can dismiss.
         final boolean showCustomizeTabView = preferenceManager.shouldShowCustomizeTabView();
         final boolean showCustomizeTabSnackBar = preferenceManager.shouldShowCustomizeTabSnackBar();
-        if (appLaunchCount < 3 || showCustomizeTabView) {
-            customizeNewTabView.setVisibility(View.VISIBLE);
-            customizeNewTabViewSnackBar.setVisibility(View.GONE);
-            if (!showCustomizeTabView) {
-                preferenceManager.setShowCustomizeTabView(true);
-            }
-            if (showCustomizeTabSnackBar) {
-                preferenceManager.setShowCustomizeTabSnackBar(false);
-            }
-        } else {
-            customizeNewTabView.setVisibility(View.GONE);
+        final Calendar installedDate = Calendar.getInstance();
+        try {
+            installedDate.setTimeInMillis(context.getPackageManager()
+                    .getPackageInfo(context.getPackageName(), 0).firstInstallTime);
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+            return;
         }
-        if (appLaunchCount >= 3 && showCustomizeTabSnackBar) {
-            customizeNewTabViewSnackBar.setVisibility(View.VISIBLE);
+        // Jan 01, 2019
+        final Calendar thresholdDate = Calendar.getInstance();
+        thresholdDate.set(2019, 1, 1);
+
+        if (installedDate.after(thresholdDate)) {
+            // New user.
+            if (appLaunchCount < 3 || showCustomizeTabView) {
```

Comment:
Please, comment the behavior. No body will understand what's happening here otherwise.